design/authenticators/authn_gcp/authn_gcp_solution_design.md

```diff
@@ -0,0 +1,600 @@
+    + [GCP Variables in Tests](#gcp-variables-in-tests)
+  * [GCE for Tests](#gce-for-tests)
+  * [Run authn-gce Requests](#run-authn-gce-requests)
+    + [Issue GCE Identity token](#issue-gce-identity-token)
+  * [Requirements from DevOps](#requirements-from-devops)
+  * [Version update](#version-update)
+- [Open questions](#open-questions)
+- [Implementation plan](#implementation-plan)
+  * [Total EE39](#total-ee39)
+    + [Ramp up EE0](#ramp-up-ee0)
+    + [Designs EE0](#designs-ee0)
+    + [POC and researches EE3](#poc-and-researches-ee3)
+    + [Implementation EE14](#implementation-ee14)
+    + [Testing EE16](#testing-ee16)
+    + [Security EE2](#security-ee2)
+    + [Docs EE4](#docs-ee4)
+  * [After GA EE6 + ??](#after-ga-ee6-----)
+    + [Designs](#designs)
+    + [Implementation](#implementation)
+    + [Testing](#testing)
+  * [Implement JWT authenticator as Beta EE18](#implement-jwt-authenticator-as-beta-ee18)
+    + [Designs](#designs-1)
+    + [Implementation](#implementation-1)
+    + [Security EE2](#security-ee2-1)
+    + [Testing](#testing-1)
+
+## Glossary
+
+| **Term** | **Description** |
+|----------|-----------------|
+|  GCP| Google Cloud Platform                 |
+|  Google compute engine (GCE)| Virtual machine in google (similar to AWS EC2)                  |
+|  Environment-provided service account| An identity for accessing private data on behalf of a service account inside Google Cloud environments |
+|  Service account key| App can self sign a google token with the service account key in order to accessing private data on behalf of a service account outside Google Cloud environments|
+
+
+
+## Useful links
+- [Feature doc](https://app.zenhub.com/workspaces/palmtree-5d99d900491c060001c85cba/issues/cyberark/conjur/1711)
+- [GCP authentication overview]( https://cloud.google.com/docs/authentication)
+- [Google metadata server](https://cloud.google.com/compute/docs/storing-retrieving-metadata#querying)
+
+## Issue description
+We want enable GCE hosts to authenticate Conjur with their google identity in order to retrieve secrets from Conjur.
+
+## Out of scope
+The only authentication strategy we are supporting is Environment-provided service account in google Compute engine.
+Other strategies are not supported: Service account key, OAuth 2.0 client, API key
+Other google components are not supported:  App Engine, GKE, Cloud Run, Cloud Functions
+
+## Solution
+We will add a new GCP authenticator to Conjur, which GCE hosts can will authenticate with by presenting their host Conjur identity and google identity as JWT token.
+
+As mentioned in the [feature doc](https://ca-il-confluence.il.cyber-ark.com/display/rndp/Conjur+GCP+Authenticator#/), before any authentication 
+request is sent to Conjur, the admin will load the authenticator policy: 
+
+```yaml
+# policy id needs to match the convention `conjur/authn-gcp/<service ID>`
+- !policy
+  id: conjur/authn-gcp/mygcp
+  body:
```

Comment:
@nessiLahav where did the `provider-uri` go? do we not have this variable anymore?